### PR TITLE
chore: tracking added when filter is added within the dashboard

### PR DESCRIPTION
**Pull request by nathaliabj**
### Pre requirements:
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/lightdash/lightdash/blob/main/.github/CONTRIBUTING.md#sending-a-pull-request).

### Description:
This PR adds tracking for when a filter is added within the dashboard, it also has a property that tells if it is on edit mode or not.

### Scope of the changes (select all that apply):
- [x] Frontend  
- [ ] Backend  
- [ ] Common  
- [ ] E2E tests  
- [ ] Docs  
- [ ] CI/CD

### CODE REVIEW

**Comment by ZeRego** [comment]
@nathaliabj I think we should add a dashboard section here so the events get getter context and for the next event a dashboardtile section

File: packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx [hunk 1]
```diff
@@ -89,6 +94,13 @@ const FilterConfiguration: FC<Props> = ({
                         internalFilterRule.values.length <= 0)
                 }
                 onClick={() => {
+                    track({
+                        name: EventName.ADD_FILTER_CLICKED,
+                        properties: {
+                            mode: isEditMode ? 'edit' : 'viewer',
+                        },
+                    });
+
```
Comment (ZeRego) [inline packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx:103]: This component is when we are editing a filter. 
Adding a filter is in packages/frontend/src/components/DashboardFilter/FilterSearch/index.tsx

Comment (nathaliabj) [reply packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx:103]: But the filter is not added until it has not been configured. do we want the event to occur when the filter is actually added, or just when clicking the `add filter` button

Comment (ZeRego) [reply packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx:103]: > But the filter is not added until it has not been configured. 
True, but that component doesn't know if the filter is being added or edited.
The `onSave` property in FilterSearch is the one that adds the filter to the context. 
```
onSave={(value) => {
                        setSelectedField(undefined);
                        addDimensionDashboardFilter(value);
                        onClose();
                    }}
                    ```

**Review by ZeRego — APPROVED** [review]

**Comment by ZeRego** [comment]
e2e tests passing here: https://github.com/lightdash/lightdash/actions/runs/1944850766

**Pull request merged**